### PR TITLE
[Data] [WIP] Large-scale data ingestion benchmark

### LINKED ISSUE #37355: [Data] Add large-scale training data ingestion benchmarks to release tests
Currently we already have single-node training data ingestion benchmarks. 

To further harden the perf, robustness and user experience, we need to extend the current benchmark to be distributed and make sure it covers all common usages. 

Basic benchmark scenario:
1. Read training/validation data from S3.
1. Apply a CPU-based preprocessing map operation.
1. Feed into the Trainer and trigger distributed training on GPU nodes.

Things to test:
- Different file input types: s3 vs local disk, raw images vs parquet files.
- Different size of input files.
- Decompress data in preprocessing.
- Different levels of shuffling: randomize_block_order + different sizes of local shuffle buffer.
- Different number of epochs. 
- Different trainers: Torch/TF.

### PR DESCRIPTION
**Pull request by scottjlee**
## Related issue number

Closes #37355

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scottjlee** [comment]
Closing in favor of https://github.com/ray-project/ray/pull/37624
